### PR TITLE
Add tasks for compiling to JS and deploying to npm

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: lib/src/npm.dart [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+/// A modifiable list of additional flags to pass to `dart2js` when compiling
+/// executables.
+var jsFlags = <String>[];
+
+/// A modifiable list of flags to pass to `dart2js` only when compiling
+/// executables in development mode.
+var jsDevFlags = <String>[];
+
+/// A modifiable list of flags to pass to `dart2js` only when compiling
+/// executables in development mode.
```
Comment (srawlins) [inline lib/src/npm.dart]: "in release mode." ?

File: lib/src/npm.dart [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+  _npmToken ??= Platform.environment["NPM_TOKEN"];
+  if (_npmToken != null) return _npmToken;
+
+  fail("pkg.npmToken must be set to deploy to npm.");
+}
+
+set npmToken(String value) => _npmToken = value;
+String _npmToken;
+
+/// Whehter [addNpmTasks] has been called yet.
```
Comment (srawlins) [inline lib/src/npm.dart]: Typo. Whether.

File: lib/src/npm.dart (hunk 1 above)
Comment (nex3) [reply lib/src/npm.dart]: Done.

File: lib/src/npm.dart (hunk 2 above)
Comment (nex3) [reply lib/src/npm.dart]: Done.

**Pull request merged**